### PR TITLE
add docs for new NuGet warning NU3043

### PR DESCRIPTION
**Pull request by kartheekp-ms**
Tracking: https://github.com/NuGet/Client.Engineering/issues/2994

Starting with .NET 9 Preview 7 and NuGet.exe 6.12, NuGet sign commands will raise a `NU3043` warning if an invalid value or SHA-1 hash is passed for the certificate fingerprint option. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).

### CODE REVIEW

**Comment by zivkan** [comment]
I disabled auto merge, because I want multiple people to have the opportunity to review and provide comments before it gets merged, not just the first person to review it 1 minute after the PR was created.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ms.topic: reference
+f1_keywords: 
+  - "NU3043"
+---
+
+# NuGet Warning NU3043
+
+`dotnet nuget sign command` - Invalid value for `--certificate-fingerprint` option. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
+
+`NuGet.exe sign command` - Invalid value for `CertificateFingerprint` option. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU3043.md:15]: This is not using the same formatting/style that most other warning pages use.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+> [!NOTE]
+> This warning will be promoted to an error around the .NET 10 timeframe.
+
+## Solution
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU3043.md:20]: All other warning pages has an `## Issue` header, that explains why the warning exists, before the solution section

Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:24]: Great catch. Fixed in the latest commit.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ms.topic: reference
+f1_keywords: 
+  - "NU3043"
+---
+
+# NuGet Warning NU3043
+
+*NuGet 6.12.0+*
+
+Invalid value for `--certificate-fingerprint` option in the `dotnet nuget sign` command or the `CertificateFingerprint` option in the `NuGet.exe sign` command. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU3043.md:15]:
```suggestion
> Invalid value for `--certificate-fingerprint` option in the `dotnet nuget sign` command or the `CertificateFingerprint` option in the `NuGet.exe sign` command. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
```

to be consistent with other warning & error pages

File: docs/reference/errors-and-warnings/NU3043.md (hunk 1 above)
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md]: I updated the docs in the latest commit. Please take a look and let me know if it looks good.

**Review by zivkan — COMMENTED** [review]
I've had these comments pending for who knows how long. I didn't re-review my comments, so I hope they're all still relevant.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+# NuGet Warning NU3043
+
+*NuGet 6.12.0+*
+
+> Invalid value for `--certificate-fingerprint` option in the `dotnet nuget sign` command or the `CertificateFingerprint` option in the `NuGet.exe sign` command. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
+>
+> [!NOTE]
+> This warning will be promoted to an error around the .NET 10 timeframe.
```
[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:18]: I get [MD028 - Blank line inside blockquote](https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md028.md) warning if I remove the `blockquote` symbol. It is also mentioned in the doc that `Some Markdown parsers will treat two blockquotes separated by one or more blank lines as the same blockquote, while others will treat them as separate blockquotes.` Given that this learn website displays two blocks separately I think we should be good.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+author: kartheekp-ms
+ms.date: 09/11/2024
+ms.topic: reference
+f1_keywords: 
+  - "NU3043"
+---
+
+# NuGet Warning NU3043
+
+*NuGet 6.12.0+*
```
[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:13]: I noticed this pattern in [NU3042](https://github.com/NuGet/docs.microsoft.com-nuget/blob/main/docs/reference/errors-and-warnings/NU3042.md?plain=1#L15) docs. I agree with your feedback. I will add `Starting with .NET 9 Preview 7 and NuGet.exe 6.12` version information to the docs.

File: docs/reference/errors-and-warnings/NU3043.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:18]: Thanks for the feedback. I removed the NOTE block.

[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:18]: Great feedback. It makes sense to me. I will update the [spec](https://github.com/NuGet/Client.Engineering/blob/main/designs/DeprecateSHA1UsageInSignCommands.md) with this decision to promote this warning to an error around .NET 11 timeframe.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+> Invalid value for `--certificate-fingerprint` option in the `dotnet nuget sign` command or the `CertificateFingerprint` option in the `NuGet.exe sign` command. The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
+>
+> [!NOTE]
+> This warning will be promoted to an error around the .NET 10 timeframe.
+
+## Issue
+
+NU3043 warning is raised when a SHA-1 certificate fingerprint is passed to the sign commands. SHA-1 hash is considered invalid due to security concerns.
```
[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md]: Fixed in the latest commit.

File: docs/reference/errors-and-warnings/NU3043.md
[the remark this answers is not in the document]
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:1]: Fixed in the latest commit.

**Review by donnie-msft — APPROVED** [review]

File: docs/reference/errors-and-warnings/NU3043.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+f1_keywords: 
+  - "NU3043"
+---
+
+# NuGet Warning NU3043
+
+> Invalid value for `--certificate-fingerprint` option in the `dotnet nuget sign` command or the `CertificateFingerprint` option in the `NuGet.exe sign` command.
+The value must be a SHA-256, SHA-384, or SHA-512 certificate fingerprint (in hexadecimal).
+
+> This warning will be promoted to an error around the `.NET 10` timeframe.
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU3043.md:16]: .NET 10 isn't a code snippet, class name, etc, so I don't think that putting it in a code block feels right.

File: docs/reference/errors-and-warnings/NU3043.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+## Issue
+
+Starting with `.NET 9 Preview 7` and `NuGet.exe 6.12`, `NU3043` warning is raised when a SHA-1 certificate fingerprint is passed to the sign commands.
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU3043.md:20]: the exact preview of .NET 9 isn't important. New previews/rcs come quickly, and soon enough .NET 9 will be GA.

And like the previous comment, .NET 9 and NuGet.exe aren't code snippets, so putting them in code blocks doesn't feel right to me. The NU code does feel fine with me, but I'd also be fine with it being taken out a code block.

**Review by zivkan — APPROVED** [review]

File: docs/reference/errors-and-warnings/NU3043.md (hunk 7 above)
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:16]: Fixed in the latest commit.

File: docs/reference/errors-and-warnings/NU3043.md (hunk 8 above)
Comment (kartheekp-ms) [reply docs/reference/errors-and-warnings/NU3043.md:20]: Fixed in the latest commit.

**Comment by dotMorten** [comment]
@kartheekp-ms Do you have any link to more information on this? The link in the issue is internal-only. I'm not quite following the security issue here, since the thumbprint it merely used to identify the certificate installed on the machine, and I can't for the life of me see how I can get a different thumbprint from my code signing certificate.

**Pull request merged**